### PR TITLE
feat: `project_id` as property

### PR DESCRIPTION
**Pull request by OmAximani0**
For now it is just configured, I will push the implementation all over the code-base on this branch only.

- fix: #123 

---

## ToDo's

> For me(@OmAximani0) only as checklist. @andrii-bodnar You can add things that should be implemented with this PR

- [x] `project_id` as optional param for all API resources
- [x] Add test for client and other API resources
- [x] Update `README.md`

### CODE REVIEW

File: crowdin_api/api_resources/abstract/resources.py [hunk 1]
```diff
@@ -5,12 +5,16 @@
...
-    def __init__(self, requester: APIRequester, page_size=25):
+    def __init__(self, requester: APIRequester, project_id: int, page_size=25):
         self.requester = requester
+        self.project_id = project_id
         self.page_size = page_size
         self._flag_fetch_all = None
         self._max_limit = None
 
+        if self.project_id is None:
+            raise ValueError("You must set project id for client")
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/abstract/resources.py:16]: The `project_id` property should not be required, as there are API methods that do not accept it.

File: crowdin_api/api_resources/abstract/resources.py [hunk 2]
```diff
@@ -5,12 +5,16 @@
 
 
 class BaseResource(metaclass=ABCMeta):
-    def __init__(self, requester: APIRequester, page_size=25):
+    def __init__(self, requester: APIRequester, project_id: int, page_size=25):
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/abstract/resources.py:8]: This configuration should be on the highest configuration level like the `TOKEN`, `ORGANIZATION` options.

Comment (andrii-bodnar) [reply crowdin_api/api_resources/abstract/resources.py:8]: Also, it should be documented in the Readme

Comment (andrii-bodnar) [reply crowdin_api/api_resources/abstract/resources.py:8]: > should be on the highest configuration level

sorry, overlooked it, it's already there

File: crowdin_api/api_resources/bundles/resource.py [hunk 3]
```diff
@@ -20,15 +20,16 @@ def get_bundles_path(self, projectId: int, bundleId: Optional[int] = None):
...
+        self, projectId: int, bundleId: int, exportId: Optional[str] = None
+    ):
         bundles_path = self.get_bundles_path(projectId, bundleId)
         if exportId:
             return f"{bundles_path}/exports/{exportId}"
         return f"{bundles_path}/exports"
 
     def list_bundles(
         self,
-        projectId: int,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/bundles/resource.py:31]: We can't change the interface of the current method that currently accepts the project ID because it'll be backward incompatible. Let's try making the `projectId` optional and populate it with the value passed to the `PROJECT_ID` option if present

**Comment by OmAximani0** [comment]
@andrii-bodnar Please checkout these [changes](https://github.com/crowdin/crowdin-api-client-python/pull/129/commits/3721f32a1f9cc0dd82800f41ce3678fa4f70294b) and I have not done changes in tests but the builds are passing please suggest changes according

File: crowdin_api/api_resources/distributions/resource.py [hunk 4]
```diff
@@ -21,7 +21,6 @@ def get_distributions_path(self, projectId: int, hash: Optional[str] = None):
 
     def list_distributions(
         self,
-        projectId: int,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/distributions/resource.py:24]: This is breaking change. The `projectId` parameter should be optional in all APIs.

File: crowdin_api/api_resources/bundles/tests/test_bundles_resources.py [hunk 5]
```diff
@@ -10,15 +10,19 @@
 
 class TestBundlesResource:
     resource_class = BundlesResource
+    PROJECT_ID = 1
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/bundles/tests/test_bundles_resources.py:13]: Let's keep the old way of passing a project ID into the methods and just add a couple of new tests with the new way of initializing project ID.

**Comment by OmAximani0** [comment]
@andrii-bodnar Please checkout below resources in which I have made the changes according to previous changes suggested and please inform me that can I change all the other API resources according to these below resources?
- [dictionaries](https://github.com/crowdin/crowdin-api-client-python/pull/129/commits/3927d1b1135ef75343ff109d49de543ede47dfb8)
- [distributions](https://github.com/crowdin/crowdin-api-client-python/pull/129/commits/62ec856998ea39fb7666c32e1098355de7688523)

**Comment by andrii-bodnar** [comment]
@OmAximani0 this looks good to me. The only thing I'd suggest changing is:

```python
projectId = projectId if projectId else self.get_project_id()
```

can be simplified:

```python
projectId = projectId or self.get_project_id()
```

**Comment by OmAximani0** [comment]
> @OmAximani0 this looks good to me. The only thing I'd suggest changing is:
> 
> ```python
> projectId = projectId if projectId else self.get_project_id()
> ```
> 
> can be simplified:
> 
> ```python
> projectId = projectId or self.get_project_id()
> ```

@andrii-bodnar Fine then, I'll implement the line as you suggested and I'll also change the other API resources according to it 👍

File: crowdin_api/api_resources/languages/tests/test_languages_resources.py [hunk 6]
```diff
@@ -3,7 +3,10 @@
 import pytest
 from crowdin_api.api_resources import LanguagesResource
 from crowdin_api.api_resources.enums import PatchOperation
-from crowdin_api.api_resources.languages.enums import LanguagesPatchPath, LanguageTextDirection
+from crowdin_api.api_resources.languages.enums import (
+    LanguagesPatchPath,
+    LanguageTextDirection,
+)
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/languages/tests/test_languages_resources.py:9]: @OmAximani0 could you please rollback all the code formatting changes that are not related to the new project ID property support? It's hard to review the PR with that amount of changes

Comment (andrii-bodnar) [reply crowdin_api/api_resources/languages/tests/test_languages_resources.py:9]: Probably, it would be better to squash all the commits as well

File: crowdin_api/api_resources/bundles/resource.py [hunk 7]
```diff
@@ -195,17 +203,19 @@ def check_bundle_export_status(
...
             path=self.get_bundles_exports_path(projectId=projectId, bundleId=bundleId, exportId=exportId),
         )
 
     def get_bundle_list_files(
         self,
-        projectId: int,
         bundleId: int,
         offset: Optional[int] = None,
         limit: Optional[int] = None,
+        projectId: Optional[int] = None,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/bundles/resource.py:218]:
```suggestion
        bundleId: int,
        projectId: Optional[int] = None,
        offset: Optional[int] = None,
        limit: Optional[int] = None,
```

**Review by andrii-bodnar — APPROVED** [review]
@OmAximani0 thank you!
